metabase/metabase
# [refs] Adding `:ident` to columns returned by `MetadataProvider`s

Closes #49686

### Description

Updates JVM, JS and unit test `MetadataProvider` implementations to include `:ident` fields for all columns.

These idents need to be globally unique and serdes portable; therefore they are not randomized.
Instead they are built from the database, schema, table and field names, eg.

```clojure
{:ident "field__Sample Database__PUBLIC__Orders__CREATED_AT"
 #_...}
```

**Note:** These `:ident` fields are not included on `/query_metadata` and other such API calls (currently).
They could be, but they can just as easily be synthesized from the DB, table and field in the JS metadata
implementation.

### How to verify

`lib.metadata` surface APIs that return columns now have checks for `:ident` on returned columns and throw if it
is not found. I have not encountered any cases where it throws, but perhaps the e2e tests will find something.

- No visible changes, but you can verify that the app is working.
- New unit tests verify that the idents we expect are returned for each `MetadataProvider` implementation

### Checklist
